lib/Basic/Version.cpp

```diff
@@ -285,6 +285,16 @@ Version::preprocessorDefinition(StringRef macroName,
   return define;
 }
 
+bool
+Version::isValidEffectiveLanguageVersion() const
+{
+  // At the moment, support setting Swift effective language version to
+  // any version starting with major number 3 or 4.
+  return Components.size() > 0 &&
+    (Components[0] == 3 ||
+     Components[0] == 4);
```

Comment:
I think we should reject "3.2", "4.1", etc. dialects that don't exist, and whitelist only those dialects that are explicitly supported.